apps/explorer/src/ui/ObjectDetails.tsx

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { ArrowUpRight16, EyeClose16, NftTypeImage24 } from '@mysten/icons';
+import { cva } from 'class-variance-authority';
+import clsx from 'clsx';
+import { useState } from 'react';
+
+import { Heading } from './Heading';
+import { ObjectLink } from './InternalLink';
+import { LightBox } from './LightBox';
+import { Text } from './Text';
+
+interface ImageProps {
+    onClick?: () => void;
+    className: string;
+    src: string;
+    blur?: boolean;
+    alt?: string;
+}
+
+const imageStyles = cva(['object-cover z-0'], {
+    variants: {
+        size: {
+            small: 'h-14 w-14',
+            large: 'h-32 w-32',
+        },
+    },
+});
+
+const textStyles = cva(['flex min-w-0 flex-col flex-nowrap'], {
+    variants: {
+        size: {
+            small: 'gap-1.25',
+            large: 'gap-2.5',
+        },
+    },
+});
+
+export interface ObjectDetailsProps {
+    id?: string;
+    image: string;
+    name?: string;
+    type: string;
+    nsfw?: boolean;
+    variant: 'small' | 'large';
+}
+
+export function ObjectDetails({
+    id,
+    image,
+    name,
+    type,
+    nsfw = false,
+    variant = 'small',
+}: ObjectDetailsProps) {
+    const [open, setOpen] = useState(false);
+    const close = () => setOpen(false);
+    const openPreview = () => setOpen(true);
+    return (
+        <div className="flex items-center gap-3.75">
+            <LightBox open={open} onClose={close}>
+                <div className="flex flex-col gap-5">
+                    <Image alt={name} src={image} className="rounded-none" />
+                    <Heading variant="heading2/semibold" color="sui-light">
+                        {name}
+                    </Heading>
+                    <Text color="gray-60" variant="body/medium">
+                        {type}
+                    </Text>
+                </div>
+            </LightBox>
+            <div className="relative">
+                <Image
+                    onClick={openPreview}
+                    alt={name}
+                    src={image}
+                    blur={nsfw}
```

Comment:
This is a behavior change I think, can you verify this is acceptable? Kevin probably has context here.

Reply:
verified with Kevin - updated UI works for sensitive images